internal/provider/resource.go

```diff
@@ -0,0 +1,13 @@
+package provider
+
+import (
+	"context"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+func DeleteContext(_ context.Context, d *schema.ResourceData, _ interface{}) diag.Diagnostics {
```

Comment:
Nitpick: would it make sense to use a name for this that doesn't clash with the field it's used in?

Reading below:

```golang
{
  // ...
  DeleteContext: DeleteContext,
  // ...
}
```

it's a bit ... odd.

Admittedly, coming up with a name can be hard (as usually, naming things always is).
How about `removeResourceFromState()`, since that's what `d.SetId("")` is for?